src/lib.rs

```diff
@@ -569,7 +582,13 @@ where
     M: ManageConnection,
 {
     fn drop(&mut self) {
-        self.pool.put_back(self.conn.take().unwrap());
+        use std::thread::panicking;
+
+        if panicking() {
```

Comment:
I'd prefer this be controlled by a configuration flag.

Reply:
How would you feel about having this controlled by a cargo feature instead? The issue is that we need a `M::Connection: UnwindSafe` bound on the impls of `UnwindSafe` and `RefUnwindSafe` if we don't do this, so having it controlled by a runtime config flag means we either can't provide those impls, or providing them is potentially unsafe.

Reply:
What's the problem with adding that extra bound?

Reply:
We don't expect it to be true the majority of the time. It is not true for `postgres::Connection` or any of the connection types in Diesel. 

We're exposing references to these types and storing them internally. If those types always get returned to the pool on drop, then we're only `UnwindSafe` if those types are. All these types use interior mutability, which is already opts them out of this trait.

So we can only be considered `UnwindSafe` if we are either not returning connections to the pool on panic, or the connections being returned are also `UnwindSafe` (and probably `RefUnwindSafe` on both impls). If this is configurable at runtime, we have to assume it's not safe to implement this. If we controlled it with a cargo feature instead, we could cfg out the impls.